src/ecdsa/mod.rs

```diff
@@ -16,11 +16,11 @@ pub use self::recovery::{RecoveryId, RecoverableSignature};
 use SECP256K1;
 
 /// An ECDSA signature
-#[derive(Copy, Clone, PartialEq, Eq)]
+#[derive(Copy, Clone, PartialEq, Eq, Hash)]
 pub struct Signature(pub(crate) ffi::Signature);
 
 /// A DER serialized Signature
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, Hash)]
```

Comment:
This will make it incompatible with the `PartialEq`, `Eq` traits, violating the required properties,
because `data` could somehow have arbitrary data after the `len` argument.

So I think we should implement Hash here manually